weldx/measurement.py

```diff
@@ -47,55 +55,459 @@ class DataTransformation:
 
     name: str
     output_signal: Signal
     error: Error
 
 
+# equipment ----------------------------------------------------------------------------
 @dataclass
+class GenericEquipment:
+    """Simple dataclass implementation for generic equipment."""
+
+    name: str
+    sources: List = field(default_factory=lambda: [])
+    data_transformations: List = field(default_factory=lambda: [])
+
+    def get_source(self, name):
+        for source in self.sources:
+            if source.name == name:
+                return source
+        raise KeyError(f"No source with name '{name}' found.")
+
+    @property
+    def source_names(self):
+        return [source.name for source in self.sources]
+
+    def get_transformation(self, name):
+        for transformation in self.data_transformations:
+            if transformation.name == name:
+                return transformation
+        raise KeyError(f"No transformation with name '{name}' found.")
+
+    @property
+    def transformation_names(self):
+        return [transformation.name for transformation in self.data_transformations]
+
+
+# DRAFT SECTION START ##################################################################
+
+# todo: - remove data from signal
+#       - factory for transformations?
+#       - which classes can be removed? -> Data
+#       - tutorial
+
+
 class MeasurementChain:
     """Simple dataclass implementation for measurement chains."""
 
-    name: str
-    data_source: Source
-    data_processors: List = field(default_factory=lambda: [])
+    def __init__(
+        self,
+        name: str,
+        source_name: str,
+        source_error: Error,
+        output_signal_type: str = None,
+        output_signal_unit: str = None,
+        output_signal: Signal = None,
+        signal_data: xr.DataArray = None,
+    ):
+        """Create a new measurement chain.
+
+        Parameters
+        ----------
+        name :
+            Name of the measurement chain
+        source_name :
+            Name of the source
+        source_error :
+            The error of the source
+        output_signal_type :
+            Type of the source's output signal (analog or digital)
+        output_signal_unit :
+            The unit of the source's output signal
+
+        """
+        from networkx import DiGraph
+
+        self._name = name
+        self._source = {
+            "name": source_name,
+            "error": source_error,
+            "equipment": None,  # This is set in a different method
+        }
+        self._prev_added_signal = None
+
+        self._graph = DiGraph()
+        self._add_signal(
+            node_id=source_name,
+            signal_type=output_signal_type,
+            unit=output_signal_unit,
+            signal=output_signal,
+        )
+        if signal_data is not None:
+            self.add_signal_data(signal_data)  # todo : test
+
+    @staticmethod
+    def construct_from_source(name, source: SignalSource):
```

Comment:
I think these should be classmethods and simply named `from_source` etc. for consistency